tests/test_models.py

```diff
@@ -86,26 +87,75 @@ def test_create_promotion(self):
         """It should Create a promotion and assert that it exists"""
-        promotion = Promotion(title="New Customer", promo_type=PromotionType.PERCENTAGE_DISCOUNT, active=True)
+        promotion = Promotion(
+            title="New Customer",
+            promo_type=PromotionType.PERCENTAGE_DISCOUNT,
+            active=True,
+        )
         self.assertEqual(str(promotion), "<Promotion New Customer id=[None]>")
         self.assertTrue(promotion is not None)
         self.assertEqual(promotion.id, None)
         self.assertEqual(promotion.title, "New Customer")
         self.assertEqual(promotion.promo_type, PromotionType.PERCENTAGE_DISCOUNT)
         self.assertEqual(promotion.active, True)
-        promotion = Promotion(title="New Customer", promo_type=PromotionType.AMOUNT_DISCOUNT, active=False)
+        promotion = Promotion(
+            title="New Customer", promo_type=PromotionType.AMOUNT_DISCOUNT, active=False
+        )
         self.assertEqual(promotion.active, False)
         self.assertEqual(promotion.promo_type, PromotionType.AMOUNT_DISCOUNT)
 
     def test_add_a_promotion(self):
         """It should Create a promotion and add it to the database"""
         promotions = Promotion.all()
         self.assertEqual(promotions, [])
-        promotion = Promotion(title="New Customer", promo_type=PromotionType.PERCENTAGE_DISCOUNT, active=True)
+        promotion = Promotion(
+            title="New Customer",
+            promo_type=PromotionType.PERCENTAGE_DISCOUNT,
+            active=True,
+        )
         self.assertTrue(promotion is not None)
         self.assertEqual(promotion.id, None)
         promotion.create()
         # Assert that it was assigned an id and shows up in the database
         self.assertIsNotNone(promotion.id)
         promotions = promotion.all()
         self.assertEqual(len(promotions), 1)
+
+    # update & delete
+    def test_update_promotion(self):
+        """It should Update a Promotion"""
+        promotion = PromotionFactory()
+        logging.debug(promotion)
+        promotion.create()
+        self.assertIsNotNone(promotion.id)
+
+        # Change it an save it
+        promotion.title = "Updated Title"
+        original_id = promotion.id
+        promotion.update()
+        self.assertEqual(promotion.id, original_id)
+        self.assertEqual(promotion.title, "Updated Title")
+
+        # Fetch it back and make sure the id hasn't changed
+        # but the data did change
+        promotions = Promotion.all()
+        self.assertEqual(len(promotions), 1)
+        self.assertEqual(promotions[0].id, original_id)
+        self.assertEqual(promotions[0].title, "Updated Title")
+
+    def test_update_no_id(self):
+        """It should not Update a Promotion with no id"""
+        promotion = PromotionFactory()
+        logging.debug(promotion)
+        promotion.id = None
+        self.assertRaises(DataValidationError, promotion.update)
+
+    def test_delete_promotion(self):
```

Comment:
If there are multiple promotions in the database, we only know that one of them is deleted, but not sure which one, should we use something like get_by_id and check if the return is null or empty?